app/src/main/java/com/example/pocketcloset/adapters/ClothingAdapter.java

```diff
@@ -0,0 +1,85 @@
+package com.example.pocketcloset.adapters;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.example.pocketcloset.R;
+import com.example.pocketcloset.models.Clothing;
+import com.parse.ParseFile;
+
+import java.util.List;
+
+public class ClothingAdapter extends RecyclerView.Adapter<ClothingAdapter.ViewHolder> {
+    private Context context;
+    private List<Clothing> clothes;
+
+    public ClothingAdapter(Context context, List<Clothing> clothes) {
+        this.context = context;
+        this.clothes = clothes;
```

Comment:
since your list is owned by the adapter, it should not also be held by the activities/fragments that create the adapter.

Basically, instead of having your Activity/Fragment do something like
```
private final List<Clothing> clothes = new ArrayList<>();

@Override
public void onCreate(Bundle savedInstanceState) {
  ...
  adapter = new ClothingAdapter(this, clothes);
  ...
}
```
I would do 
```
@Override
public void onCreate(Bundle savedInstanceState) {
  ...
  adapter = new ClothingAdapter(this, new ArrayList<Clothing>());
  ...
}
```

This makes it very plain that the Activity/Fragment should never modify the list. All modifications should go through the Adapter.